### PR TITLE
ui/hooks: drop down menu

### PR DESCRIPTION
**Pull request by KatieMSB**
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**
Convert DropDownMenu.js to use hooks.

**Which issue(s) this PR fixes:**
Part of #923

### CODE REVIEW

File: web/src/app/dialogs/components/DropDownMenu.js [hunk 1]
```diff
@@ -1,65 +1,61 @@
...
+        onClose={() => {
+          setAnchorEl(null)
+        }}
+      >
+        {options.map((option) => (
+          <MenuItem
+            key={option.label}
+            onClick={() => {
+              setAnchorEl(null)
+              option.onClick()
```
Comment (mastercactapus) [inline web/src/app/dialogs/components/DropDownMenu.js:40]: This was intended to be fired upon `onExited`, after animations have finished. IIRC we had a lot of test flakiness leading to it being implemented this way.

Might require a little creativity to get it to work with hooks. Or some careful thought/validation to see if it's still required.

**Review by mastercactapus — CHANGES_REQUESTED** [review]

File: web/src/app/dialogs/components/DropDownMenu.js (hunk 1 above)
Comment (Forfold) [reply web/src/app/dialogs/components/DropDownMenu.js:40]: I'd be curious to see how this runs against our tests as is, without the use of classes/old setState

File: web/src/app/dialogs/components/DropDownMenu.js [hunk 2]
```diff
@@ -1,65 +1,61 @@
...
-    )
-  }
+DropDownMenu.propTypes = {
+  options: p.arrayOf(
+    p.shape({
+      label: p.string,
+      onClick: p.func,
+    }),
+  ),
+  onExited: p.func,
```
Comment (Forfold) [inline web/src/app/dialogs/components/DropDownMenu.js:58]: I don't think this is needed, but we can likely extend `Menu` when we convert this to TypeScript and have a spread anywho

**Review by Forfold — DISMISSED** [review]

File: web/src/app/dialogs/components/DropDownMenu.js (hunk 1 above)
Comment (mastercactapus) [reply web/src/app/dialogs/components/DropDownMenu.js:40]: IIRC it was a focus-stealing issue -- dialog opening before the menu closed or something like that

Comment (mastercactapus) [reply web/src/app/dialogs/components/DropDownMenu.js:40]: vague memories of some nightmarish test debugging come to mind

Comment (KatieMSB) [reply web/src/app/dialogs/components/DropDownMenu.js]: This component isn't being utilized anywhere but I ran through some cypress scenarios locally and everything seemed to work as expected with the latest changes.

**Review by Forfold — CHANGES_REQUESTED** [review]
Since you mentioned that this isn't really being utilized anywhere, what do you think of just deleting this file and updating the once reference we have to it (snippet below) in `DialogTitleWrapper` to use `OtherActions` from utils instead? It's the more recent version we've been using that should handle switching between mobile and desktop.

**`web/src/app/dialogs/components/DialogTitleWrapper.js`**
```
  let menu
  if (options && options.length > 0 && fullScreen) {
    menu = <DropDownMenu options={options} color='white' />
  } else if (options && options.length > 0) {
    menu = (
      <div className={classes.topRightActions}>
        <DropDownMenu options={options} />
      </div>
    )
  }
```

**Comment by KatieMSB** [comment]
> Since you mentioned that this isn't really being utilized anywhere, what do you think of just deleting this file and updating the once reference we have to it (snippet below) in `DialogTitleWrapper` to use `OtherActions` from utils instead? It's the more recent version we've been using that should handle switching between mobile and desktop.
> 
> **`web/src/app/dialogs/components/DialogTitleWrapper.js`**
> 
> ```
>   let menu
>   if (options && options.length > 0 && fullScreen) {
>     menu = <DropDownMenu options={options} color='white' />
>   } else if (options && options.length > 0) {
>     menu = (
>       <div className={classes.topRightActions}>
>         <DropDownMenu options={options} />
>       </div>
>     )
>   }
> ```

Yea that makes sense, I can make a separate pr for that and go ahead and close this one if that sounds good.

**Comment by Forfold** [comment]
Sure, whatever is easiest for you! I wouldn't mind staying on this PR + updating the title either

**Comment by KatieMSB** [comment]
Closing in favor of #1156

**Pull request closed without merging**